ArduCopter/control_safe_rtl.cpp

```diff
@@ -0,0 +1,136 @@
+#include "Copter.h"
+
+/*
+ * Init and run calls for Safe_RTL flight mode
+ *
+ * This code uses the SafeRTL path that is already in memory, and feeds it into WPNav, one point at a time.
+ * Once the copter is close to home, it will run a standard land controller.
+ */
+
+bool Copter::safe_rtl_init(bool ignore_checks)
+{
+    if ((position_ok() || ignore_checks) && g2.safe_rtl.is_active()) {
+        // initialise waypoint and spline controller
+        wp_nav->wp_and_spline_init();
+
+        // set current target to a reasonable stopping point
+        Vector3f stopping_point;
+        pos_control->get_stopping_point_xy(stopping_point);
+        pos_control->get_stopping_point_z(stopping_point);
+        wp_nav->set_wp_destination(stopping_point);
+
+        // initialise yaw to obey user parameter
+        set_auto_yaw_mode(get_default_auto_yaw_mode(true));
+
+        // wait for cleanup of return path
+        safe_rtl_state = SafeRTL_WaitForPathCleanup;
+        return true;
+    } else {
+        return false;
+    }
+}
+
+// perform cleanup required when leaving safe_rtl
+void Copter::safe_rtl_exit()
+{
+    g2.safe_rtl.cancel_request_for_thorough_cleanup();
+}
+
+void Copter::safe_rtl_run()
+{
+    switch(safe_rtl_state){
```

Comment:
In ardupilot, instead of "){", put whitespace in between.

Reply:
yup, will fix!